x-pack/plugins/apm/public/components/app/service_groups/service_group_save/select_services.tsx

```diff
@@ -78,6 +76,19 @@ export function SelectServices({
     }
   }, [isEdit, serviceGroup.kuery]);
 
+  useEffect(() => {
+    if (!stagedKuery) {
+      return;
+    }
+    const { isValid, isParsingError, message } =
+      validateServiceGroupKuery(stagedKuery);
+    if (isValid || isParsingError) {
+      setKueryValidationMessage('');
+    } else {
+      setKueryValidationMessage(message);
+    }
```

Comment:
If we combine `isValid` and `isParsingError`:

```suggestion
    setKueryValidationMessage(isValid ? '' : message)
```

Reply:
Actually, it shouldn't matter. `message` is already set to `` when `isValid=true`

```suggestion
      setKueryValidationMessage(message);
```